src/plugins/dashboard/public/application/top_nav/editor_menu.tsx

```diff
@@ -0,0 +1,210 @@
+}
+
+interface FactoryGroup {
+  id: string;
+  appName: string;
+  icon: IconType;
+  panelId: number;
+  factories: EmbeddableFactoryDefinition[];
+}
+
+export const EditorMenu = ({ dashboardContainer, createNewVisType }: Props) => {
+  const {
+    embeddable,
+    visualizations,
+    usageCollection,
+  } = useKibana<DashboardAppServices>().services;
+
+  const trackUiMetric = usageCollection?.reportUiCounter.bind(
+    usageCollection,
+    DashboardConstants.DASHBOARDS_ID
+  );
+
+  const createNewAggsBasedVis = useCallback(
+    async () =>
+      visualizations.showNewVisModal({
+        originatingApp: DashboardConstants.DASHBOARDS_ID,
+        outsideVisualizeApp: true,
+        showAggsSelection: true,
+      }),
+    [visualizations]
+  );
+
+  const getVisTypesByGroup = (group: VisGroups) =>
+    visualizations
+      .getByGroup(group)
+      .sort(({ name: a }: BaseVisType | VisTypeAlias, { name: b }: BaseVisType | VisTypeAlias) => {
+        if (a < b) {
+          return -1;
+        }
+        if (a > b) {
+          return 1;
+        }
+        return 0;
+      })
+      .filter(({ hidden }: BaseVisType) => !hidden);
+
+  const promotedVisTypes = getVisTypesByGroup(VisGroups.PROMOTED);
+  const visTypeAliases = visualizations
+    .getAliases()
+    .sort(({ promotion: a = false }: VisTypeAlias, { promotion: b = false }: VisTypeAlias) =>
+      a === b ? 0 : a ? -1 : 1
+    );
+
+  const factories = embeddable
+    ? Array.from(embeddable.getEmbeddableFactories()).filter(
+        ({ type, isEditable, canCreateNew, isContainerType }) =>
+          isEditable() && !isContainerType && canCreateNew() && type !== 'visualization'
+      )
+    : [];
+
+  const factoryGroupMap: Record<string, FactoryGroup> = {};
+  const ungroupedFactories: EmbeddableFactoryDefinition[] = [];
+  let panelCount = 1;
+
+  factories.forEach((factory: EmbeddableFactoryDefinition, index) => {
+    const { grouping } = factory;
+
+    if (grouping) {
+      grouping.forEach((group) => {
+        if (factoryGroupMap[group.id]) {
+          factoryGroupMap[group.id].factories.push(factory);
+        } else {
+          factoryGroupMap[group.id] = {
+            id: group.id,
+            appName: group.getDisplayName ? group.getDisplayName({ embeddable }) : group.id,
+            icon: (group.getIconType ? group.getIconType({ embeddable }) : 'empty') as IconType,
+            factories: [factory],
+            panelId: panelCount,
+          };
+
+          panelCount++;
+        }
+      });
+    } else {
+      ungroupedFactories.push(factory);
+    }
+  });
+
+  const getVisTypeMenuItem = (visType: BaseVisType): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, titleInWizard, description, icon = 'empty' } = visType;
+
+    return {
+      name: titleInWizard || title,
+      icon: icon as string,
+      onClick: createNewVisType(visType),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
```

Comment:
I notice that the `Aggregation based` type does not have a tooltip, and also does not have a right arrow icon that would indicate that it opens a modal

Reply:
I've updated the menu, and now the `Aggregation based` opens a sub-menu and has an arrow on the right.

<img width="266" alt="Screen Shot 2021-04-14 at 6 18 19 PM" src="https://user-images.githubusercontent.com/1697105/114800061-d5cda980-9d4d-11eb-8b66-f8e5457cba38.png">
<img width="278" alt="Screen Shot 2021-04-14 at 6 06 59 PM" src="https://user-images.githubusercontent.com/1697105/114800063-d6fed680-9d4d-11eb-827e-be85c0365dbb.png">